### PR TITLE
feat: make gloss approval more discoverable

### LINKED ISSUE #282: make gloss approval more discoverable.
**Is your feature request related to a problem? Please describe.**
In all user interviews I've done, users aren't discovering that glosses can be approved and unapproved. Additionally, there is no way to approve/unapprove glosses on a mobile device (including tablets) since we can't listen for Enter on them.

**Describe the solution you'd like**
One option is to add a button that's not in the tab sequence after each input to either approve or unapprove the gloss. We can additionally only make it appear when the input is focused, and also put the keyboard shortcut in a tooltip.

### PR DESCRIPTION
**Pull request by Pertempto**
## What has changed

Added approve and revoke approval buttons.

## Connected Issues

Closes #282 

### Approve Button
1. Go to the translate page.
2. Focus an unapproved gloss input that contains content
    - A green approval button should appear
    - It should not be in the tab sequence
    - It should have a localized tooltip
    - Clicking it should approve the gloss

### Revoke Button
1. Go to the translate page.
2. Focus an approved gloss input
    - A red revoke button should appear
    - It should not be in the tab sequence
    - It should have a localized tooltip
    - Clicking it should remove approval from the gloss

Also test with machine generated glosses, to make sure that the icon is positioned correctly.
## Post-Deployment

- [ ] Nothing required

### CODE REVIEW

**Comment by Pertempto** [comment]
# To Do

- [x] Fix styling of machine gloss icon

File: packages/web/src/public/locales/en/translate.json [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
+  "approve_tooltip": "Approve (Enter)",
   "approved": "Approved",
   "interlinear": "Interlinear",
   "next_verse": "Next Verse",
   "previous_verse": "Previous Verse",
   "saving": "saving",
   "select_verse": "Select verse",
-  "strongs": "Strongs"
+  "strongs": "Strongs",
+  "unapprove_tooltip": "Unapprove (Esc)"
```
Comment (Pertempto) [inline packages/web/src/public/locales/en/translate.json]: Should we use a different term? "Unapprove" isn't a real word, so it doesn't translate well.

**Comment by Pertempto** [comment]
Ok, this should be ready. I think the machine gloss icon is positioned correctly now.

File: packages/web/src/public/locales/en/translate.json (hunk 1 above)
Comment (arrocke) [reply packages/web/src/public/locales/en/translate.json]: Let's use "Revoke" instead, that's really what this is

**Review by arrocke — CHANGES_REQUESTED** [review]
This is looking pretty good! I'm seeing three issues yet and then this can merge:
* Focusing the word shifts everything around slightly
* Button doesn't appear if machine gloss is shown
* Clicking the button should return focus to the input

**Comment by Pertempto** [comment]
It should be fixed now.

File: packages/web/src/features/translation/TranslateWord.tsx [hunk 2]
```diff
@@ -137,12 +137,19 @@ const TranslateWord = forwardRef<TranslateWordRef, TranslateWordProps>(
               dir={targetLanguage?.textDirection ?? TextDirection.LTR}
             >
               <div className="group-focus-within/input-row:block hidden">
-                {status === 'saved' && (
+                {glossValue && status !== 'approved' && (
```
Comment (arrocke) [inline packages/web/src/features/translation/TranslateWord.tsx]: Can we remove this? Right now if you have an empty gloss, and you type something, the button doesn't appear until you move off and come back
```suggestion
                {status !== 'approved' && (
```

**Review by arrocke — APPROVED** [review]

File: packages/web/src/features/translation/TranslateWord.tsx (hunk 2 above)
Comment (Pertempto) [reply packages/web/src/features/translation/TranslateWord.tsx]: Then the approval button shows up for a completely empty input, which can't be approved.

Comment (Pertempto) [reply packages/web/src/features/translation/TranslateWord.tsx]: I added a `currentInputValue` state variable to help here. It seems to work well, but there might be edge cases I didn't catch.

Comment (Pertempto) [reply packages/web/src/features/translation/TranslateWord.tsx]: Can you try it out before we merge?

**Pull request merged**